pkg/mdns/mdns.go

```diff
@@ -167,14 +159,31 @@ func (m *MDNS) BrowseMDNS() {
 			hostCustomDomain := m.ReplaceLocal(entry.Host)
 			srvName := strings.SplitN(m.ReplaceLocal(entry.Name), ".", 2)[1]
 			entry.Host = hostCustomDomain
-			(*m.srvHosts)[srvName] = append((*m.srvHosts)[srvName], entry)
+			srvHosts[srvName] = append(srvHosts[srvName], entry)
 		}
 	}()
 
 	mdns.Lookup("_workstation._tcp", entriesCh)
 	close(entriesCh)
 	mdns.Lookup("_etcd-server-ssl._tcp", srvEntriesCh)
 	close(srvEntriesCh)
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	// Clear maps so we don't have stale entries
+	for k := range *m.mdnsHosts {
+		delete(*m.mdnsHosts, k)
+	}
+	for k := range *m.srvHosts {
+		delete(*m.srvHosts, k)
```

Comment:
I wonder if it wouldn't be better to have the mDNS hold pointers to the maps and then:
1. lock
2. point to the new map
3. unlock

Reply:
That would be nice, but the value receiver for ServeDNS seems to screw us up again. Because we get a copy of the plugin object, changing what the pointers point to in BrowseMDNS doesn't actually get reflected in ServeDNS. I'm not _entirely_ sure why, but there must be a copy made in the plugin infrastructure somewhere that disconnects us from the instance we create in setup.